akka-docs/src/main/paradox/cluster-client.md

```diff
@@ -227,3 +232,62 @@ This can be useful when initial contacts are provided from some kind of service
 are entirely dynamic and the entire cluster might shut down or crash, be restarted on new addresses. Since the
 client will be stopped in that case a monitoring actor can watch it and upon `Terminate` a new set of initial
 contacts can be fetched and a new cluster client started.
+
+## Migration to Akka gRPC
+
+Cluster Client is deprecated and it is not advised to build new applications with Cluster Client.
+We recommend using [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html) over using Cluster Client.
+
+Cluster Client is convenient to use if you have Akka actors on both client and server side, but
+a more decoupled, and therefore better, solution would be to use Akka gRPC and define
+application specific protocol buffer messages and gRPC service calls. The benefits would be:
+
+* Easier to update clients and servers independent of each other.
+* More well defined protocol between client and server.
+* More secure solution by using TLS for gRPC (HTTP/2) instead of exposing Akka Remoting outside the Akka Cluster.
+* The clients don't even have to be using Akka.
+* See also [gRPC vs Akka Remoting](https://doc.akka.io/docs/akka-grpc/current/whygrpc.html#grpc-vs-akka-remoting)
+
+Existing users of Cluster Client may migrate directly to plain Akka gRPC, or via a migration
+step that requires less re-write of the application. That migration step is described here but we recommend
+against using the approach for new applications. New applications should use plain Akka gRPC.
+
+An example is provided to illustrate an approach to migrate from the deprecated Cluster Client to Akka gRPC,
+with minimal changes to your existing code. The example is intended to be copied and adjusted to your needs.
+It will not be provided as a published artifact.
+
+* [akka-samples/akka-sample-cluster-cluster-client-grpc-scala](https://github.com/akka/akka-samples/tree/2.6/akka-sample-cluster-cluster-client-grpc-scala) implemented in Scala
+* [akka-samples/akka-sample-cluster-cluster-client-grpc-java](https://github.com/akka/akka-samples/tree/2.6/akka-sample-cluster-cluster-client-grpc-java) implemented in Java
+
+The example is still using an actor on the client side to have an API that is very close
+to the original Cluster Client. The messages this actor can handle corresponds to the
+@ref:[Distributed Pub Sub](distributed-pub-sub.md) messages on the server side, such as
+`ClusterClient.Send` and `ClusterClient.Publish`.
+
+The `ClusterClient` actor delegates those messages to the gRPC client and on the
+server side those are translated and delegated to the destination actors that
+are registered via the `ClusterClientReceptionist` in the same way as in the original.
+
+The application specific messages are wrapped and serialized with Akka serialization.
+
+### Differences
+
+Apart from that the underlying implementation is using gRPC instead of Actor messages
+and Akka Remoting it's worth pointing out the following differences.
```

Comment:
Maybe make it more clear that the differences below is about the migration-step above and not about Akka gRPC in general?

"... it's worth pointing out the following differences with this Cluster Client emulation style Akka gRPC usage." 

or something along those lines.